### PR TITLE
feat: add kernel-bazzite to cache

### PR DESCRIPTION
**Pull request by antheas**
Early draft PR for adding kernel-bazzite to kernel-cache. Waiting for Kyle to comment for it. Kind of untested.

### CODE REVIEW

**Review by KyleGospo — APPROVED** [review]

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -144,6 +147,9 @@ jobs:
                 $dnf copr enable -y sentry/kernel-ba
                 linux=$($dnf repoquery --repoid copr:copr.fedorainfracloud.org:sentry:kernel-ba --whatprovides kernel | sort -V | tail -n1 | sed 's/.*://')
                 ;;
+              "bazzite")
+                latest="$(curl "https://api.github.com/repos/hhd-dev/kernel-bazzite/releases/latest" )"
+                linux=$(echo -E "$latest" | jq -r '.assets[].name' | grep -E 'kernel-.*.rpm' | grep "fc${{ matrix.fedora_version }}" | head -1 | sed "s/kernel-//g" | sed "s/.rpm//g" )
               "surface")
```
Comment (p5) [inline .github/workflows/reusable-build.yml:153]:
```suggestion
                linux=$(echo -E "$latest" | jq -r '.assets[].name' | grep -E 'kernel-.*.rpm' | grep "fc${{ matrix.fedora_version }}" | head -1 | sed "s/kernel-//g" | sed "s/.rpm//g" )
                ;;
              "surface")
```

**Comment by antheas** [comment]
really did not like that json passthrough. Switched to a tag and hardcoded the links

**Comment by antheas** [comment]
Should be optional and empty if unset, let me fix it.

**Review by m2Giles — APPROVED** [review]

**Review by p5 — APPROVED** [review]
LGTM

**Pull request merged**